### PR TITLE
Logical `OR` for relationship MATCH

### PR DESCRIPTION
**Pull request by jackboyla**
supports `MATCH (n)-[r:colleague|friend]->(m)`
plus some tests for it

```python
from grandcypher import GrandCypher
import networkx as nx

host = nx.MultiDiGraph()
host.add_node("a", name="Alice", age=25)
host.add_node("b", name="Bob", age=30)
host.add_node("c", name="Christine", age=30)
host.add_node("d", name="Derek", age=57)
host.add_edge("a", "b", __labels__={"friend"}, years=1, friendly="very")
host.add_edge("b", "a", __labels__={"colleague"}, years=2)
host.add_edge("b", "a", __labels__={"mentor"}, years=4)
host.add_edge("b", "c", __labels__={"chef"}, years=12)
host.add_edge("d", "b", __labels__={"friend"}, years=1, friendly="mildly")

qry = """
MATCH (n)-[r:colleague|friend]->(m)
RETURN n.name, m.name, r.__labels__, r.years, r.friendly
"""
res = GrandCypher(host).run(qry)
print(res)

'''
{
    'n.name': ['Alice', 'Bob', 'Derek'], 
    'm.name': ['Bob', 'Alice', 'Bob'], 
    'r.__labels__': [
        {0: {'friend'}}, {0: {'colleague'}, 1: {'mentor'}}, {0: {'friend'}}
    ], 
    'r.years': [{0: 1}, {0: 2, 1: 4}, {0: 1}], 
    'r.friendly': [{0: 'very'}, {0: None, 1: None}, {0: 'mildly'}]
}
'''
```

### CODE REVIEW

**Comment by j6k4m8** [comment]
I can't quite tell for my phone but I'll look at this might need to be rebased on latest trunk? 

this is so cool — I had no idea cypher could do this!

**Comment by jackboyla** [comment]
you're right! I just rebased 😄 yeah it's kind of niche and not really necessary (could just use `WHERE`) but it's nice to have

File: grandcypher/test_queries.py [hunk 1]
```diff
@@ -1661,3 +1661,94 @@ def test_path(self, graph_type):
...
+    def test_match_with_single_or_operator(self, graph_type):
+        host = graph_type()
+        host.add_node("a", name="Alice")
+        host.add_node("b", name="Bob")
+        host.add_node("c", name="Carol")
+        host.add_edge("a", "b", __labels__={"LOVES"})
+        host.add_edge("b", "c", __labels__={"WORKS_WITH"})
+
+        qry = """
+        MATCH (n1)-[r:LOVES|WORKS_WITH]->(n2)
```
Comment (j6k4m8) [inline grandcypher/test_queries.py:1677]: Alice and Bob should disclose this to HR

**Review by j6k4m8 — APPROVED** [review]

**Comment by j6k4m8** [comment]
totally, this is super nice to have! and the closer we can get to feature parity with the cypher spec the better, imo! every feature is great :)

File: grandcypher/test_queries.py (hunk 1 above)
Comment (jackboyla) [reply grandcypher/test_queries.py:1677]: 😂

**Pull request merged**